src/main/java/org/manifold/compiler/front/FunctionTypeValueExpression.java

```diff
@@ -0,0 +1,63 @@
+package org.manifold.compiler.front;
+
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.Value;
+
+public class FunctionTypeValueExpression extends Expression {
+
+  private Expression inputTypeExpr;
+  public Expression getInputTypeExpression() {
+    return inputTypeExpr;
+  }
+  
+  private Expression outputTypeExpr;
+  public Expression getOutputTypeExpression() {
+    return outputTypeExpr;
+  }
+  
+  public FunctionTypeValueExpression(
+      Expression inputType, Expression outputType) {
+    this.inputTypeExpr = inputType;
+    this.outputTypeExpr = outputType;
+  }
+
+  @Override
+  public TypeValue getType(Scope scope) {
+    // TODO Auto-generated method stub
+    return null;
+  }
+
+  @Override
+  public Value getValue(Scope scope) {
+    // TODO Auto-generated method stub
+    return null;
```

Comment:
I'm going to leave this for both because we "both know" that this class is going away eventually (and fixing this requires a type-check that I don't want to do at this point).